plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,186 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binary
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/anchore/stereoscope"
+	"github.com/openclarity/vmclarity/plugins/runner/internal/runtimehandler"
+	"github.com/openclarity/vmclarity/plugins/runner/types"
+	"github.com/openclarity/vmclarity/plugins/sdk-go/plugin"
+	"github.com/openclarity/vmclarity/scanner/utils/containerrootfs"
+	"github.com/openclarity/vmclarity/utils/command"
+)
+
+type binaryRuntimeHandler struct {
+	config types.PluginConfig
+
+	ready                bool
+	commandState         *command.State
+	pluginServerEndpoint string
+}
+
+func New(ctx context.Context, config types.PluginConfig) (runtimehandler.PluginRuntimeHandler, error) {
+	return &binaryRuntimeHandler{
+		config: config,
+	}, nil
+}
+
+func (h *binaryRuntimeHandler) Start(ctx context.Context) error {
+	pluginDir := filepath.Join(os.TempDir(), "vmclarity-plugins", h.config.Name)
+
+	image, err := stereoscope.GetImage(ctx, h.config.ImageName)
+	if err != nil {
+		return fmt.Errorf("unable to get image(%s): %w", h.config.ImageName, err)
+	}
+
+	if _, err := os.Stat(pluginDir); os.IsNotExist(err) {
+		err = containerrootfs.ToDirectory(ctx, h.config.ImageName, pluginDir)
+		if err != nil {
+			return fmt.Errorf("unable to extract image(%s): %w", h.config.ImageName, err)
+		}
+	}
+
+	// Mount input from host
+	// /tmp/vmclarity-plugins/kics + /input + /host-dir-to-scan
+	inputDirMountPoint := filepath.Join(pluginDir, runtimehandler.RemoteScanInputDirOverride, h.config.InputDir)
+	err = os.MkdirAll(inputDirMountPoint, 0755)
```

Comment:
input dir that needs to be scanned should be mounted in read-only mode

Reply:
This is done.